modules/nlu/src/backend/language-provider.ts

```diff
@@ -166,23 +166,18 @@ export class RemoteLanguageProvider implements LanguageProvider {
     return { validProvidersCount: this._validProvidersCount, validLanguages: Object.keys(this.langs) }
   }
 
-  private async queryProvider<T>(lang: string, path: string, body: any, returnProperty: string): Promise<T> {
-    const providers = this.langs[lang]
-
-    if (!providers) {
+  private getAvailableProviders(lang: string): Gateway[] {
+    if (!this.langs[lang]) {
       throw new Error(`Language "${lang}" is not supported by the configured language sources`)
     }
 
-    for (const provider of providers) {
-      if (provider.disabledUntil > new Date()) {
-        debug('source disabled, skipping', {
-          source: provider.source,
-          errors: provider.errors,
-          until: provider.disabledUntil
-        })
-        continue
-      }
+    return this.langs[lang].filter(x => !x.disabledUntil || x.disabledUntil <= new Date())
+  }
 
+  private async queryProvider<T>(lang: string, path: string, body: any, returnProperty: string): Promise<T> {
```

Comment:
Do I understand that we query each available providers for any query ?

Reply:
No, only the first one that returns successfully